core/vtk/ttkDataSetToTable/ttkDataSetToTable.cpp

```diff
@@ -1,124 +1,72 @@
-        cerr << "[ttkDataSetToTable] Error: input has no point data." << endl;
-        return -1;
-      }
-#endif
-
-#ifndef TTK_ENABLE_KAMIKAZE
-      const SimplexId numberOfArrays = inputPointData->GetNumberOfArrays();
-      if(numberOfArrays <= 0) {
-        cerr << "[ttkDataSetToTable] Error: input point data is empty." << endl;
-        return -1;
-      }
-#endif
-
-      table->GetRowData()->ShallowCopy(inputPointData);
-    } break;
-
-    case AssociationType::Cell: {
-      vtkCellData *inputCellData = input->GetCellData();
-#ifndef TTK_ENABLE_KAMIKAZE
-      if(!inputCellData) {
-        cerr << "[ttkDataSetToTable] Error: input has no cell data." << endl;
-        return -1;
-      }
-#endif
-
-#ifndef TTK_ENABLE_KAMIKAZE
-      const SimplexId numberOfArrays = inputCellData->GetNumberOfArrays();
-      if(numberOfArrays <= 0) {
-        cerr << "[ttkDataSetToTable] Error: input cell data is empty." << endl;
-        return -1;
-      }
-#endif
-
-      table->GetRowData()->ShallowCopy(inputCellData);
-    } break;
-
-    case AssociationType::Field: {
-      vtkFieldData *inputFieldData = input->GetFieldData();
-#ifndef TTK_ENABLE_KAMIKAZE
-      if(!inputFieldData) {
-        cerr << "[ttkDataSetToTable] Error: input has no field data." << endl;
-        return -1;
-      }
-#endif
-
-#ifndef TTK_ENABLE_KAMIKAZE
-      const SimplexId numberOfArrays = inputFieldData->GetNumberOfArrays();
-      if(numberOfArrays <= 0) {
-        cerr << "[ttkDataSetToTable] Error: input field data is empty." << endl;
-        return -1;
-      }
-#endif
-
-      table->GetRowData()->ShallowCopy(inputFieldData);
-    } break;
+int ttkDataSetToTable::FillOutputPortInformation(int port,
+                                                 vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkTable");
+    return 1;
   }
-
-  output->ShallowCopy(table);
-
-  {
-    stringstream msg;
-    msg << "[ttkDataSetToTable] Memory usage: " << m.getElapsedUsage() << " MB."
-        << endl;
-    dMsg(cout, msg.str(), memoryMsg);
-  }
-
   return 0;
 }
 
 int ttkDataSetToTable::RequestData(vtkInformation *request,
                                    vtkInformationVector **inputVector,
                                    vtkInformationVector *outputVector) {
-  Memory m;
+  ttk::Timer t;
 
-  vtkDataSet *input = vtkDataSet::GetData(inputVector[0]);
-  vtkTable *output = vtkTable::GetData(outputVector);
+  std::string targetAttributeName
+    = this->DataAssociation == 0
+        ? "Point"
+        : this->DataAssociation == 1 ? "Cell" : "Field";
 
-  doIt(input, output);
+  this->printMsg("Converting " + targetAttributeName + "Data to Table", 0, 0,
+                 ttk::debug::LineMode::REPLACE);
 
-  {
-    stringstream msg;
-    msg << "[ttkDataSetToTable] Memory usage: " << m.getElapsedUsage() << " MB."
-        << endl;
-    dMsg(cout, msg.str(), memoryMsg);
+  auto input = vtkDataSet::GetData(inputVector[0]);
```

Comment:
you may want to check input is not nullptr here, don't you?

Reply:
alright, forget about it. we'll do a global pass on those at a later stage.

Reply:
Yes, I will add a sanity check.